### ISSUE TITLE
[BUG] Passing argument implementing interface using GSON

### ISSUE THREAD

**Issue by junhuhdev** [jobrunr/jobrunr#79]
**Describe the bug**

Trying to enqueue a class implementing an interface

```java
BackgroundJob.enqueue(() -> pipeline.send(command));
```

Command is a class implementing an interface. 
```java
@NoArgsConstructor
@AllArgsConstructor
@Data
public class CreateRequest implements Command<CreateRequest .Response> {
	private String ssn;
	private String msisdn;
}
```

**Environment**
I'm using version 1.2.3
I'm running on JRE / JDK  java 15
I'm using the following StorageProvider PostgresSQL

**To Reproduce**
Provide a simple project or code fragment which shows how to reproduce the issue.
```java
BackgroundJob.enqueue(() -> pipeline.send(command));
```

```java
2020-12-23 19:50:20.545  WARN 5188 --- [pool-1-thread-2] org.jobrunr.server.JobZooKeeper          : JobRunr encountered a problematic exception. Please create a bug report (if possible, provide the code to reproduce this and the stacktrace) - Processing will continue.

java.lang.RuntimeException: Unable to invoke no-args constructor for interface an.awesome.pipelinr.Command. Registering an InstanceCreator with Gson for this type may fix this problem.
	at com.google.gson.internal.ConstructorConstructor$14.construct(ConstructorConstructor.java:228) ~[gson-2.8.6.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:212) ~[gson-2.8.6.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[gson-2.8.6.jar:na]
	at org.jobrunr.utils.mapper.gson.RuntimeClassNameTypeAdapterFactory$1.read(RuntimeClassNameTypeAdapterFactory.java:219) ~[jobrunr-1.2.3.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[gson-2.8.6.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[gson-2.8.6.jar:na]
	at org.jobrunr.utils.mapper.gson.RuntimeClassNameTypeAdapterFactory$1.read(RuntimeClassNameTypeAdapterFactory.java:219) ~[jobrunr-1.2.3.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[gson-2.8.6.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[gson-2.8.6.jar:na]
	at org.jobrunr.utils.mapper.gson.RuntimeClassNameTypeAdapterFactory$1.read(RuntimeClassNameTypeAdapterFactory.java:219) ~[jobrunr-1.2.3.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[gson-2.8.6.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:932) ~[gson-2.8.6.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:1003) ~[gson-2.8.6.jar:na]
	at com.google.gson.internal.bind.TreeTypeAdapter$GsonContextImpl.deserialize(TreeTypeAdapter.java:162) ~[gson-2.8.6.jar:na]
	at org.jobrunr.utils.mapper.gson.JobParameterDeserializer.deserializeToObject(JobParameterDeserializer.java:22) ~[jobrunr-1.2.3.jar:na]
	at org.jobrunr.utils.mapper.gson.JobParameterDeserializer.deserialize(JobParameterDeserializer.java:18) ~[jobrunr-1.2.3.jar:na]
	at org.jobrunr.utils.mapper.gson.JobParameterDeserializer.deserialize(JobParameterDeserializer.java:12) ~[jobrunr-1.2.3.jar:na]
	at com.google.gson.internal.bind.TreeTypeAdapter.read(TreeTypeAdapter.java:69) ~[gson-2.8.6.jar:na]
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41) ~[gson-2.8.6.jar:na]
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:82) ~[gson-2.8.6.jar:na]
	at com.google.gson.internal.bind.CollectionTypeAdapterFactory$Adapter.read(CollectionTypeAdapterFactory.java:61) ~[gson-2.8.6.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[gson-2.8.6.jar:na]
	at org.jobrunr.utils.mapper.gson.RuntimeClassNameTypeAdapterFactory$1.read(RuntimeClassNameTypeAdapterFactory.java:219) ~[jobrunr-1.2.3.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[gson-2.8.6.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[gson-2.8.6.jar:na]
	at org.jobrunr.utils.mapper.gson.RuntimeClassNameTypeAdapterFactory$1.read(RuntimeClassNameTypeAdapterFactory.java:219) ~[jobrunr-1.2.3.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[gson-2.8.6.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[gson-2.8.6.jar:na]
	at org.jobrunr.utils.mapper.gson.RuntimeClassNameTypeAdapterFactory$1.read(RuntimeClassNameTypeAdapterFactory.java:219) ~[jobrunr-1.2.3.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[gson-2.8.6.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131) ~[gson-2.8.6.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222) ~[gson-2.8.6.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[gson-2.8.6.jar:na]
	at org.jobrunr.utils.mapper.gson.RuntimeClassNameTypeAdapterFactory$1.read(RuntimeClassNameTypeAdapterFactory.java:219) ~[jobrunr-1.2.3.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[gson-2.8.6.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[gson-2.8.6.jar:na]
	at org.jobrunr.utils.mapper.gson.RuntimeClassNameTypeAdapterFactory$1.read(RuntimeClassNameTypeAdapterFactory.java:219) ~[jobrunr-1.2.3.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[gson-2.8.6.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[gson-2.8.6.jar:na]
	at org.jobrunr.utils.mapper.gson.RuntimeClassNameTypeAdapterFactory$1.read(RuntimeClassNameTypeAdapterFactory.java:219) ~[jobrunr-1.2.3.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[gson-2.8.6.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131) ~[gson-2.8.6.jar:na]
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222) ~[gson-2.8.6.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[gson-2.8.6.jar:na]
	at org.jobrunr.utils.mapper.gson.RuntimeClassNameTypeAdapterFactory$1.read(RuntimeClassNameTypeAdapterFactory.java:219) ~[jobrunr-1.2.3.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[gson-2.8.6.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[gson-2.8.6.jar:na]
	at org.jobrunr.utils.mapper.gson.RuntimeClassNameTypeAdapterFactory$1.read(RuntimeClassNameTypeAdapterFactory.java:219) ~[jobrunr-1.2.3.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[gson-2.8.6.jar:na]
	at com.google.gson.TypeAdapter.fromJsonTree(TypeAdapter.java:285) ~[gson-2.8.6.jar:na]
	at org.jobrunr.utils.mapper.gson.RuntimeClassNameTypeAdapterFactory$1.read(RuntimeClassNameTypeAdapterFactory.java:219) ~[jobrunr-1.2.3.jar:na]
	at com.google.gson.TypeAdapter$1.read(TypeAdapter.java:199) ~[gson-2.8.6.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:932) ~[gson-2.8.6.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:897) ~[gson-2.8.6.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:846) ~[gson-2.8.6.jar:na]
	at com.google.gson.Gson.fromJson(Gson.java:817) ~[gson-2.8.6.jar:na]
	at org.jobrunr.utils.mapper.gson.GsonJsonMapper.deserialize(GsonJsonMapper.java:69) ~[jobrunr-1.2.3.jar:na]
	at org.jobrunr.jobs.mappers.JobMapper.deserializeJob(JobMapper.java:20) ~[jobrunr-1.2.3.jar:na]
	at org.jobrunr.storage.sql.common.JobTable.toJob(JobTable.java:226) ~[jobrunr-1.2.3.jar:na]
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195) ~[na:na]
	at org.jobrunr.storage.sql.common.db.SqlSpliterator.tryAdvance(SqlSpliterator.java:44) ~[jobrunr-1.2.3.jar:na]
	at java.base/java.util.Spliterator.forEachRemaining(Spliterator.java:326) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474) ~[na:na]
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913) ~[na:na]
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[na:na]
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578) ~[na:na]
	at org.jobrunr.storage.sql.common.JobTable.selectJobsByState(JobTable.java:144) ~[jobrunr-1.2.3.jar:na]
	at org.jobrunr.storage.sql.common.DefaultSqlStorageProvider.getJobs(DefaultSqlStorageProvider.java:132) ~[jobrunr-1.2.3.jar:na]
	at org.jobrunr.storage.ThreadSafeStorageProvider.getJobs(ThreadSafeStorageProvider.java:117) ~[jobrunr-1.2.3.jar:na]
	at org.jobrunr.server.JobZooKeeper.checkForEnqueuedJobs(JobZooKeeper.java:149) ~[jobrunr-1.2.3.jar:na]
	at org.jobrunr.server.JobZooKeeper.run(JobZooKeeper.java:72) ~[jobrunr-1.2.3.jar:na]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset$$$capture(FutureTask.java:305) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:832) ~[na:na]
Caused by: java.lang.UnsupportedOperationException: Interface can't be instantiated! Interface name: an.awesome.pipelinr.Command
	at com.google.gson.internal.UnsafeAllocator.assertInstantiable(UnsafeAllocator.java:117) ~[gson-2.8.6.jar:na]
	at com.google.gson.internal.UnsafeAllocator$1.newInstance(UnsafeAllocator.java:49) ~[gson-2.8.6.jar:na]
	at com.google.gson.internal.ConstructorConstructor$14.construct(ConstructorConstructor.java:225) ~[gson-2.8.6.jar:na]
	... 78 common frames omitted

```

**Comment by junhuhdev** [comment]
I guess the fix would be to be able to set a custom jobMapper & jsonMapper. 

Currently useStorageProvider always set default jobMapper.

**Comment by rdehuyss** [comment]
Solved in release v1.2.4.
